quic/s2n-quic/Cargo.toml

```diff
@@ -15,7 +15,7 @@ default = [
     "provider-address-token-default",
     "provider-tls-default",
 ]
-
+provider-fips-crypto = ["s2n-quic-crypto/fips"]
```

Comment:
Why `provider`? This feature seems different than the others in that its not changing out an actual provider

Reply:
I think it would be better to "extend" the s2n provider to enable fips? Something like `provider-tls-s2n-fips`?

Reply:
its not just TLS that changes with this flag though?

Reply:
Looking at the other features in `s2n-quic`:
- `provider-address-token-default`: brings in dependencies needed for the default address token provider
- `provider-tls-default`: enables the default TLS provider based on platform detection
- `provider-event-tracing`: enabled tracing provider by default

They all have to do with an existing provider, but for fips, its not actually changing anything about any of the providers, thats why I wasn't sure `provider` makes sense in the name here.

Reply:
On a related note: running into some [usage issues](https://github.com/aws/s2n-quic/compare/ak-fipsSecurityPolicy?expand=1#diff-60bc3691ceca61d5071aaf2d020cff0f50be1977d3f95bc5b5e93cda4b8be8fbR13-R24) with `provider-tls-default` and `fips`.

I listed 2 options, neither of which seem amazing.

---

I have a slight preference for "provider" since `fips` affects the behavior of the tls provider. All our current features are also [prefixed](https://github.com/aws/s2n-quic/blob/main/quic/s2n-quic/Cargo.toml#L13-L32) with "provider".

Reply:
Discussed offline and "provider-crypto-fips" seems like a good fit. 'fips' touches more than just tls so that doesnt feel right. We dont have a crypto provider today but its possible to image that we could have one in the future so this naming would fit with configuring cypto functionality.

Thoughts @camshaft